### ISSUE TITLE
Issues when maximizing a window with MaxWidth constrain

### ISSUE THREAD

**Issue by leonidcan** [AvaloniaUI/Avalonia#14167]
## Describe the bug
When windows has MaxWidth constrain, maximizing it creates visual artefacts. For example title bar hangs over the window and bottom part of the window is hidden behind the task bar on Windows

## To Reproduce

1. Add MaxWidth="200" to <Window...> control 
2. Run app (Windows platform)
3. Click Maximize button
4. See the issues with window's caption bar and Windows task bar (when it's located at the bottom of the desktop)

## Expected behavior
Window should look normal and not hidden under the taskbar

## Environment 

- OS: Windows
- Avalonia-Version: 11 latest

## Additional context
N/A

**Comment by leonidcan** [comment]
![image](https://github.com/AvaloniaUI/Avalonia/assets/18264548/62215cb8-fbfe-48f1-bac7-ae1600b59217)

**Comment by TomEdwardsEnscape** [comment]
I'm not seeing that on master with Windows 10. Are you using Windows 11? You didn't specify in the bug report.

![image](https://github.com/AvaloniaUI/Avalonia/assets/109803929/f3514fd1-80fd-48de-b8dd-37a62fb179e6)

**Comment by timunie** [comment]
A minimal repro could also be useful as well as specifying the OS

**Comment by leonidcan** [comment]
Yes it's Windows 11, I already don't have minimal repro, I will try to reproduce when I have time.
